src/commcare_cloud/ansible/plugins/lookup/cached_aws_secret.py

```diff
@@ -1,27 +1,34 @@
+import json
+import os
+import cPickle
+import random
+import time
+
 from ansible.plugins.lookup import aws_secret
+from cryptography.fernet import Fernet, InvalidToken
 
 __metaclass__ = type
 
+
 DOCUMENTATION = aws_secret.DOCUMENTATION
 RETURN = aws_secret.RETURN
 
 
 class LookupModule(aws_secret.LookupModule):
-    _secrets_cache = {}
 
     def run(self, terms, variables, **kwargs):
+        terms = tuple(terms)
+        term, = terms
+        value = self.get_cache(term, inventory_dir=variables['inventory_dir'])
         try:
-            terms = tuple(terms)
-            if terms not in self._secrets_cache:
+            if value is Ellipsis:
                 print('Fetching', terms)
                 try:
                     value = super(LookupModule, self).run(terms, variables, **kwargs)
                 except Exception as e:
                     print('Caching error: ', e)
                     value = e
-                self._secrets_cache[terms] = value
-
-            value = self._secrets_cache[terms]
+                self.set_cache(term, value, inventory_dir=variables['inventory_dir'])
```

Comment:
Why save the exception in the cache?

Reply:
The most common exception is that the secret doesn't exist. I want to cache that so that next time the same secret is asked for in the same ansible run, we already know it doesn't exist and don't need to make another API request